driver/csiplugin/connectors/rest_v2.go

```diff
@@ -594,6 +608,86 @@ func (s *SpectrumRestV2) CreateFileset(ctx context.Context, filesystemName strin
 	return nil
 }
 
+func (s *SpectrumRestV2) SetBucketKeys(ctx context.Context, bucketInfo map[string]string) error {
+	loggerID := utils.GetLoggerId(ctx)
+	klog.V(4).Infof("[%s] rest_v2 SetBucketKeys", loggerID)
+
+	keyreq := SetBucketKeysRequest{}
+
+	keyreq.BucketName = bucketInfo[BucketName]
+	keyreq.AccessKey = bucketInfo[bucketAccesskey]
+	keyreq.SecretKey = bucketInfo[bucketSecretkey]
+
+	// Extract the hostname without the port
+	parsedURL, err := url.Parse(bucketInfo[BucketEndpoint])
+	if err != nil {
+		return fmt.Errorf("failed to parse endpoint URL %s, error %v", bucketInfo[BucketEndpoint], err)
+	}
+	hostname := parsedURL.Hostname()
+	keyreq.Server = hostname
+
+	setBucketKeysURL := "scalemgmt/v2/bucket/keys"
+	setBucketKeysResponse := GenericResponse{}
+
+	err = s.doHTTP(ctx, setBucketKeysURL, "PUT", &setBucketKeysResponse, keyreq)
+	if err != nil {
+		klog.Errorf("[%s] Failed to set keys for the bucket %s, error: %v", loggerID, bucketInfo[BucketName], err)
+		return err
+	}
+
+	err = s.isRequestAccepted(ctx, setBucketKeysResponse, setBucketKeysURL)
+	if err != nil {
+		klog.Errorf("[%s] The set keys request is not accepted for processing for the bucket %s, error: %v", loggerID, bucketInfo[BucketName], err)
+		return err
+	}
+
+	err = s.WaitForJobCompletion(ctx, setBucketKeysResponse.Status.Code, setBucketKeysResponse.Jobs[0].JobID)
+	if err != nil {
+		klog.Errorf("[%s] Failed to set keys for the bucket %s, error: %v", loggerID, bucketInfo[BucketName], err)
+		return err
+	}
+	return nil
+}
+
+func (s *SpectrumRestV2) CreateS3CacheFileset(ctx context.Context, filesystemName string, filesetName string, mode string, opts map[string]interface{}, bucketInfo map[string]string) error {
+	loggerID := utils.GetLoggerId(ctx)
+	klog.V(4).Infof("[%s] rest_v2 CreateS3CacheFileset. filesystem: %s, fileset: %s, mode: %s, opts: %v", loggerID, filesystemName, filesetName, mode, opts)
+
+	filesetreq := CreateS3CacheFilesetRequest{}
+	filesetreq.FilesetName = filesetName
+	filesetreq.UseObjectFs = true
+	filesetreq.Mode = mode
+
+	filesetreq.Endpoint = bucketInfo[BucketEndpoint]
+	filesetreq.BucketName = bucketInfo[BucketName]
+
+	createFilesetURL := fmt.Sprintf("scalemgmt/v2/filesystems/%s/filesets/cos", filesystemName)
+	createFilesetResponse := GenericResponse{}
+
+	err := s.doHTTP(ctx, createFilesetURL, "POST", &createFilesetResponse, filesetreq)
+	if err != nil {
+		klog.Errorf("[%s] Failed to create an AFM cache fileset %s, error: %v", loggerID, filesetName, err)
```

Comment:
Should we avoid using AFM in the logs as we are not exposing AFM to end user? This error message could be modified as : 

Failed to create cache volume (filesetName) with mode (mode)?

Reply:
This is not the final cache volume creation step, this is the actual fileset creation step.